# CoExShiny
CoExShiny is an R package designed to facilitate the creation of interactive Shiny web applications. Its main objective is to enable the visualization of co-expression patterns between two genes using single cell data input in Seurat format. The application offers multiple formats for data representation, including umap, bar plots, stacked bar plots, and tables. The package also allows for the display of metadata plots and summaries in formats that are most suitable for the data. 

# installation 

##### required packages
```R
# checking and installing required packages
reqPkg = c("devtools", "Seurat", "SeuratObject", "dplyr", "glue", "shiny",
     "tidyr", "shinythemes", "DT" , "ggplot2", "tibble", "RColorBrewer")
newPkg = reqPkg[!(reqPkg %in% installed.packages()[,"Package"])]
if(length(newPkg)){install.packages(newPkg)}
```

##### installing and loading CoExShiny
```R
devtools::install_github("rahatghvi/CoExShiny")

library(CoExShiny)
```

# Usage guide
#### Creating the app
CoExShiny receives input in the form of clustered Seurat objects. It processes the input to create a list of all markers that exhibit differential expression and a corresponding metadata list. These files are then used to generate a customized Shiny app. The folder will be created in current working directory. 

```R
data("neural")

# you can specify the directory path where you want to create the files to ensure they are created in a specific folder.
setwd("D:/USER")

# Default folder name is CoEx
create_coex_files(neural)
```


Another code is then used to run the app. Alternatively the app can be find under CoEx/bin/app folder and it can be launched via opening opening either `server.R` or `ui.R`  in Rstudio and clicking "Run App" button at top right side of source window

```R
# replace 'D:/USER/CoEx' with the absolute path to the newly created folder
CoExApp("D:/USER/CoEx")
```

#### Using the app
##### MetaData:

The first panel of the interface displays a plot and provides a summary of metadata information.

![Pasted image 20240318145806](https://github.com/rahatghvi/CoExShiny/assets/153312046/614d0215-533b-4d5c-a871-17c6a9efb53a)

##### Markers:
In the second panel, users can generate a list of genes from a data set of differentially expressed markers in the dataset. 
It is important to note that users must save the gene list by clicking the "save" button, for the list to be able to proceed to the third panel.

![Pasted image 20240318152609](https://github.com/rahatghvi/CoExShiny/assets/153312046/186cf6be-9ecd-441b-a03d-3c479f455298)

#### Figures
In the third panel, it is required to select two genes, and upon doing so, a table of gene summary statistics will be generated by clicking the "generate gene summary" button.

![Pasted image 20240318154551](https://github.com/rahatghvi/CoExShiny/assets/153312046/92ba4acc-1144-444f-b597-6b5c399176fa)

After, an expression threshold must be specified for each gene. By clicking the "generate plot" button, plots and tables will be generated. It is important to note that the coexpression level of each gene is classified as either positive (P) or negative (N), where coexpression of PN implies that the first gene is expressed, and the second gene is not. 

Three types of plots, namely, UMAP plot, bar plot, and stacked bar plot, will be generated along with a table. If required, images of either of these plots or tables can be downloaded by clicking the download button located at the end of each panel.

![Untitled Diagram drawio](https://github.com/rahatghvi/CoExShiny/assets/153312046/18eb584d-f293-49c8-a6d5-da535c637978)

Furthermore, in the "Download Subset" panel, it is possible to download a particular subset by selecting the clusters and coexpression type of interest.

# References:
default data included is sourced from Yang et al. 2021:
Yang, A. C. et al. 2021. Dysregulation of brain and choroid plexus cell types in severe COVID-19. Nature 595(7868), pp. 565-571. doi: 10.1038/s41586-021-03710-0

